### PR TITLE
framework/bluetooth: Add bluetooth capi module

### PR DESCRIPTION
**Pull request by Wootak**

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cccc66c39c3b0fa438b05b1c48ee0095719a5c44] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cccc66c39c3b0fa438b05b1c48ee0095719a5c44] - Code Rule Check OK.

File: external/bluetooth/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,129 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
```
Comment (pillip8282) [inline external/bluetooth/CMakeLists.txt]: is this file necessary?

File: external/bluetooth/Make.defs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+###########################################################################
+#
+# Copyright 2017 Samsung Electronics All Rights Reserved.
+#
```
Comment (pillip8282) [inline external/bluetooth/Make.defs]: I think 2019 is right

File: external/bluetooth/capi-network-bluetooth.pc.in [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Package Information for pkg-config
+
```
Comment (pillip8282) [inline external/bluetooth/capi-network-bluetooth.pc.in]: is this file used?

File: external/bluetooth/packaging/capi-network-bluetooth-devel.manifest [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<manifest>
```
Comment (pillip8282) [inline external/bluetooth/packaging/capi-network-bluetooth-devel.manifest]: is this file used?

File: external/bluetooth/packaging/capi-network-bluetooth.spec [hunk 5]
```diff
@@ -0,0 +1,108 @@
+Name:       capi-network-bluetooth
+Summary:    Network Bluetooth Framework
```
Comment (pillip8282) [inline external/bluetooth/packaging/capi-network-bluetooth.spec]: is this file used?

File: external/bluetooth/test/bt_unit_test.c [hunk 6]
```diff
@@ -0,0 +1,9442 @@
...
+	{"bt_adapter_set_manufacturer_data_changed_cb"
+		, BT_UNIT_TEST_FUNCTION_ADAPTER_SET_MANUFACTURER_DATA_CHANGED_CB},
+	{"bt_adapter_unset_manufacturer_data_changed_cb"
+		, BT_UNIT_TEST_FUNCTION_ADAPTER_UNSET_MANUFACTURER_DATA_CHANGED_CB},
+	{"bt_adapter_set_manufacturer_data"
+		, BT_UNIT_TEST_FUNCTION_ADAPTER_SET_MANUFACTURER_DATA},
+	{"bt_adapter_set_authentication_req_cb"
+		, BT_UNIT_TEST_FUNCTION_ADAPTER_SET_AUTHENTICATION_REQUSET_CB},
+	{"bt_adapter_unset_authentication_req_cb"
+		, BT_UNIT_TEST_FUNCTION_ADAPTER_UNSET_AUTHENTICATION_REQUSET_CB},
```
Comment (pillip8282) [inline external/bluetooth/test/bt_unit_test.c]: Are you going to have BT test with this file? if not you'd better not to include it

File: external/bluetooth/CMakeLists.txt (hunk 1 above)
Comment (Wootak) [reply external/bluetooth/CMakeLists.txt]: Not necessary in TizenRT. But, I just copy this file from external package as like other module.
Should I remove unnecessary files?

File: external/bluetooth/Make.defs (hunk 2 above)
Comment (Wootak) [reply external/bluetooth/Make.defs]: Thank you. I will fix.

File: external/bluetooth/adaptation/include/bt-adaptation-common.h [hunk 7]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2011 Samsung Electronics Co., Ltd All Rights Reserved
```
Comment (NilabhShant) [inline external/bluetooth/adaptation/include/bt-adaptation-common.h]: Please make 2011 to 2019.

File: external/bluetooth/adaptation/include/bt-adaptation-device.h [hunk 8]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2011 Samsung Electronics Co., Ltd All Rights Reserved
```
Comment (NilabhShant) [inline external/bluetooth/adaptation/include/bt-adaptation-device.h]: same as above.

File: external/bluetooth/adaptation/include/bt-adaptation-gatt.h [hunk 9]
```diff
@@ -0,0 +1,420 @@
+/*
+ * Copyright (c) 2011 Samsung Electronics Co., Ltd All Rights Reserved
```
Comment (NilabhShant) [inline external/bluetooth/adaptation/include/bt-adaptation-gatt.h]: same as above.

File: external/bluetooth/adaptation/include/bt-adaptation-gatt.h [hunk 10]
```diff
@@ -0,0 +1,420 @@
...
+
+int bt_adapt_gatt_disconnect(const char *address, int client_id);
+
+/*******************************
+ * GATT Client Function Header *
+********************************/
+
+int bt_adapt_gatt_client_init(const char *address, int *client_id);
+
+int bt_adapt_gatt_client_deinit(	int client_id);
```
Comment (NilabhShant) [inline external/bluetooth/adaptation/include/bt-adaptation-gatt.h]: Please remove tab/spaces before "int client_id"

File: external/bluetooth/src/bluetooth-adapter.c [hunk 11]
```diff
@@ -0,0 +1,2311 @@
+/*
+ * Copyright (c) 2011 Samsung Electronics Co., Ltd All Rights Reserved
```
Comment (NilabhShant) [inline external/bluetooth/src/bluetooth-adapter.c]: Please make 2011 to 2019 and please check this for all file headers.

File: external/bluetooth/adaptation/include/bt-adaptation-common.h (hunk 7 above)
Comment (Wootak) [reply external/bluetooth/adaptation/include/bt-adaptation-common.h]: These files are external module. and, I didn't change the code.
So, it is better to do not change the boilerplate.

File: external/bluetooth/src/bluetooth-adapter.c (hunk 11 above)
Comment (Wootak) [reply external/bluetooth/src/bluetooth-adapter.c]: Same as above.

File: external/bluetooth/adaptation/include/bt-adaptation-gatt.h (hunk 9 above)
Comment (Wootak) [reply external/bluetooth/adaptation/include/bt-adaptation-gatt.h]: same as above.

File: external/bluetooth/adaptation/include/bt-adaptation-device.h (hunk 8 above)
Comment (Wootak) [reply external/bluetooth/adaptation/include/bt-adaptation-device.h]: same as above.

File: external/bluetooth/test/bt_unit_test.c (hunk 6 above)
Comment (Wootak) [reply external/bluetooth/test/bt_unit_test.c]: I will remove test file.

File: external/bluetooth/capi-network-bluetooth.pc.in (hunk 3 above)
Comment (Wootak) [reply external/bluetooth/capi-network-bluetooth.pc.in]: I will remove unnecessary files.

File: external/bluetooth/packaging/capi-network-bluetooth-devel.manifest (hunk 4 above)
Comment (Wootak) [reply external/bluetooth/packaging/capi-network-bluetooth-devel.manifest]: same as above.

File: external/bluetooth/packaging/capi-network-bluetooth.spec (hunk 5 above)
Comment (Wootak) [reply external/bluetooth/packaging/capi-network-bluetooth.spec]: same as above.

**Comment by seinfra** [comment]
Target : [d1c10dcce204a0f0674708736b51579fb5b3ac73] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d1c10dcce204a0f0674708736b51579fb5b3ac73] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [41a1b89ca795bc66f05f57cbd8263e0522cd82c2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [41a1b89ca795bc66f05f57cbd8263e0522cd82c2] - Code Rule Check OK.

File: external/bluetooth/adaptation/bt-adatation-gatt.c [hunk 12]
```diff
@@ -0,0 +1,315 @@
...
+********************************/
+
+
+int bt_adapt_gatt_client_init(const char *address, int *client_id)
+{
+	/* To be implemented */
+	return BT_ERROR_NONE;
+}
+
+int bt_adapt_gatt_client_deinit(	int client_id)
```
Comment (drashti304) [inline external/bluetooth/adaptation/bt-adatation-gatt.c]: Please remove extra space within the bracket

**Comment by seinfra** [comment]
Target : [fbb2aaf85c7c2bb7e0af6488dc18eeb970ad32b8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fbb2aaf85c7c2bb7e0af6488dc18eeb970ad32b8] - Code Rule Check Result:
framework/src/bluetooth/adaptation/bt-adatation-adapter.c:164: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/bluetooth/adaptation/bt-adatation-gatt.c:92: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/bluetooth/adaptation/include/bt-adaptation-adapter.h:86: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/bluetooth/src/bluetooth-otp.c:63: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/bluetooth/src/bluetooth-otp.c:63: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/bluetooth/src/bluetooth-otp.c:64: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/bluetooth/src/bluetooth-otp.c:64: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [c8f92a5886e437a34b51b20e11cf63df3eeed6d0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c8f92a5886e437a34b51b20e11cf63df3eeed6d0] - Code Rule Check OK.
